### PR TITLE
Nested folders: very naive implementations of move/delete

### PR DESCRIPTION
**Pull request by ashharrison90**
**What is this feature?**

- very naive implementation of move/delete
- there are **many** bugs that are known and not worth fixing in this implementation 😂 
- follow up PR will refactor `getChildren` logic so it can all be managed by RTK query

**Why do we need this feature?**

- get move/delete working e2e

**Who is this feature for?**

- anyone using nested folders

**Which issue(s) does this PR fix?**:

For https://github.com/grafana/grafana/issues/65681
For https://github.com/grafana/grafana/issues/65679

**Special notes for your reviewer:**

Please check that:
- [ ] It works as expected from a user's perspective.
- [ ] If this is a pre-GA feature, it is behind a feature toggle.
- [ ] The docs are updated, and if this is a [notable improvement](https://grafana.com/docs/writers-toolkit/writing-guide/contribute-release-notes/#how-to-determine-if-content-belongs-in-a-whats-new-document), it's added to our [What's New](https://grafana.com/docs/writers-toolkit/writing-guide/contribute-release-notes/) doc.

### CODE REVIEW

**Comment by joshhunt** [comment]
> there are many bugs that are known

@ashharrison90 Don't leave me hanging - what are they? 😅  Might be worth just listing in the PR so we know what's happening.

**Comment by ashharrison90** [comment]
@joshhunt haha. i would summarise them all as being to do with expanded folders. for example:
- if you delete something from an expanded folder, it won't be removed from the UI
- if you move something into/out of an expanded folder, it won't be updated in the UI
- once you've confirmed your move/delete, your checkbox selection still remains
  - maybe we should fix this one, have a `clearSelection` action which is called afterwards? 🤔 

all of these bugs can be worked around by manually refreshing the page. the real solution is moving these calls into RTK query, but that will probably take a bit of time.

File: public/app/features/browse-dashboards/components/BrowseActions/BrowseActions.tsx [hunk 1]
```diff
@@ -6,51 +6,99 @@ import { Button, useStyles2 } from '@grafana/ui';
...
+    onActionComplete?.();
+  };
 
-  const onMove = () => {
+  const showMoveModal = () => {
     appEvents.publish(
       new ShowModalReactEvent({
         component: MoveModal,
         props: {
           selectedItems,
```
Comment (joshhunt) [inline public/app/features/browse-dashboards/components/BrowseActions/BrowseActions.tsx:77]: I'm not a fan of the modal events system, and rather we just rendered the modal components ourselves. But that's not an issue of _this_ PR so its fine.

**Review by joshhunt — APPROVED** [review]
Looks good. Not totally sure about the right way to do the RTK Query mutations with observing _all_ of their statuses. I guessing it might involve moving the delete/move loops into a queryFn so from the component we can observe it as a single action.

**Pull request merged**